gui/src/components/firmware-update/FirmwareUpdate.tsx

```diff
@@ -280,25 +281,29 @@ export function FirmwareUpdate() {
         {}
       ),
     });
-    queueFlashing(devices);
+    setStatus({});
+    clear();
   };
 
   const startUpdate = () => {
     const selectedDevices = Object.keys(selectedDevicesForm)
       .filter((d) => selectedDevicesForm[d])
```

Comment:
can you add this check inside the reduce at least, it will reduce needed iterations